Analysis/Tasks/PWGCF/dptdptcorrelations.cxx

```diff
@@ -0,0 +1,407 @@
+enum SystemType {
+  kNoSystem = 0, ///< no system defined
+  kpp,           ///< **p-p** system
+  kpPb,          ///< **p-Pb** system
+  kPbp,          ///< **Pb-p** system
+  kPbPb,         ///< **Pb-Pb** system
+  kXeXe,         ///< **Xe-Xe** system
+  knSystems      ///< number of handled systems
+};
+
+/* probably this will not work in the context of distributed processes */
+/* we need to revisit it!!!!                                           */
+SystemType fSystem = kNoSystem;
+TH1F* fhCentMultB = nullptr;
+TH1F* fhCentMultA = nullptr;
+TH1F* fhVertexZB = nullptr;
+TH1F* fhVertexZA = nullptr;
+TH1F* fhPtB = nullptr;
+TH1F* fhPtA = nullptr;
+TH1F* fhPtPosB = nullptr;
+TH1F* fhPtPosA = nullptr;
+TH1F* fhPtNegB = nullptr;
+TH1F* fhPtNegA = nullptr;
+
+TH1F* fhEtaB = nullptr;
+TH1F* fhEtaA = nullptr;
+
+TH1F* fhPhiB = nullptr;
+TH1F* fhPhiA = nullptr;
+
+TH2F* fhEtaVsPhiB = nullptr;
+TH2F* fhEtaVsPhiA = nullptr;
+
+TH2F* fhPtVsEtaB = nullptr;
+TH2F* fhPtVsEtaA = nullptr;
+
+SystemType getSystemType()
+{
+  /* we have to figure out how extract the system type */
+  return kPbPb;
+}
+
+bool IsEvtSelected(aod::CollisionEvSelCent const& collision)
+{
+  if (collision.alias()[kINT7]) {
+    if (collision.sel7()) {
+      if (zvtxlow < collision.posZ() and collision.posZ() < zvtxup) {
+        return true;
+      }
+      return false;
+    }
+  }
+  return false;
+}
+
+bool matchTrackType(aod::TrackData const& track)
+{
+  switch (cfgTrackType) {
+    case 1:
+      if (track.isGlobalTrack() != 0)
+        return true;
+      else
+        return false;
+      break;
+    case 2:
+      if (track.isGlobalTrackSDD() != 0)
+        return true;
+      else
+        return false;
+      break;
+    default:
+      return false;
+  }
+}
+
+std::tuple<uint8_t, uint8_t> AcceptTrack(aod::TrackData const& track)
+{
+
+  uint8_t asone = DPTDPT_FALSE;
+  uint8_t astwo = DPTDPT_FALSE;
+
+  /* TODO: incorporate a mask in the scanned tracks table for the rejecting track reason */
+  if (matchTrackType(track)) {
+    if (ptlow < track.pt() and track.pt() < ptup and etalow < track.eta() and track.eta() < etaup) {
+      if (((track.charge() > 0) and (cfgTrackOneCharge > 0)) or ((track.charge() < 0) and (cfgTrackOneCharge < 0))) {
+        asone = DPTDPT_TRUE;
+      } else if (((track.charge() > 0) and (cfgTrackTwoCharge > 0)) or ((track.charge() < 0) and (cfgTrackTwoCharge < 0))) {
+        astwo = DPTDPT_TRUE;
+      }
+    }
+  }
+  return std::make_tuple(asone, astwo);
+}
+} /* end namespace dptdptcorrelations */
+
+// Task for <dpt,dpt> correlations analysis
+// FIXME: this should really inherit from AnalysisTask but
+//        we need GCC 7.4+ for that
+
+using namespace dptdptcorrelations;
+
+struct DptDptCorrelationsFilterAnalysisTask {
+
+  OutputObj<TDirectory> fOutput{"DptDptCorrelationsGlobalInfo", OutputObjHandlingPolicy::AnalysisObject};
+
+  OutputObj<TH1F> fOutCentMultB{"CentralityB", OutputObjHandlingPolicy::AnalysisObject};
```

Comment:
You can do a full declaration of the histogram here, and then you don't need any line in init(). For example:
OutputObj<TH2F> etaphiH{TH2F("etaphi", "etaphi", 100, 0., 2. * M_PI, 102, -2.01, 2.01)};

Reply:
Yes, you are right. I used that way in the QA task.
I'm trying few things I'm not sure they will work in the grid/train scenarios and this will be probably one of them.
The plan is, anyway, to use the HistogramRegistry I guess